### PR TITLE
Bug 1522899 - Limit spocs per domain

### PR DESCRIPTION
**Pull request by ScottDowne**
To test:

1. Use pref dev-test-all
2. Load a new tab
3. Start scrolling down the page

Expected: spocs should be diverse and re ordered a bit.

### CODE REVIEW

File: test/unit/lib/DiscoveryStreamFeed.test.js [hunk 1]
```diff
@@ -440,6 +440,100 @@ describe("DiscoveryStreamFeed", () => {
...
+        ],
+      });
+
+      assert.equal(result.spocs[0].score, 0.9);
+    });
+    it("should filter out duplicate campigns", () => {
+      const result = feed.transform({
+        spocs: [
+          {campaign_id: 2, item_score: 0.8, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
```
Comment (punamdahiya) [inline test/unit/lib/DiscoveryStreamFeed.test.js]: we can change item_score here to 0.6 to test campaign_id with the higher score is picked first

File: test/unit/lib/DiscoveryStreamFeed.test.js [hunk 2]
```diff
@@ -440,6 +440,100 @@ describe("DiscoveryStreamFeed", () => {
...
+        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
+        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
+        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
+        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
+        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
+        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
+      ]);
+    });
+  });
+
```
Comment (punamdahiya) [inline test/unit/lib/DiscoveryStreamFeed.test.js]: trying for this case with item_scores below is failing test, 

      const result = feed.transform({
        spocs: [
          {campaign_id: 2, item_score: 0.8, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
          {campaign_id: 2, item_score: 0.9, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.8, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
        ],
      });

      assert.deepEqual(result.spocs, [
        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
        {campaign_id: 1, item_score: 0.8, score: 0.9, min_score: 0.1},
        {campaign_id: 2, item_score: 0.9, score: 0.8, min_score: 0.1},
        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
      ]);

Comment (punamdahiya) [reply test/unit/lib/DiscoveryStreamFeed.test.js]: it was my miss, passing with above values, will good to have test updated with different item_scores to cover that

**Comment by ScottDowne** [comment]
@punamdahiya Updated!

**Pull request merged**